File: Enclosure/src/Enclosure.java
import java.util.*;
import java.io.*;

public class Enclosure{
	public static ArrayList<Integer> getPositions(String operations){
		ArrayList<Integer> answers=new ArrayList<Integer>();
		Boolean open=false;
		Integer position = null;
		for(int i=0; i<operations.length(); i++){
			if(operations.charAt(i)=='('){
				open=true;
				position=i;
				break;
			}else if(operations.charAt(i)==')'){
				position=i;
				break;
			}
		}
		if(open){
			for(int j=position+2; j<operations.length(); j++){
				if(operations.charAt(j)!='+'&&operations.charAt(j)!='-'&&operations.charAt(j)!='/'&&operations.charAt(j)!='*'){
					answers.add(j+2);
				}
			}
		}else{
			for(int j=position-2; j>=0; j--){
				if(operations.charAt(j)!='+'&&operations.charAt(j)!='-'&&operations.charAt(j)!='/'&&operations.charAt(j)!='*'){
					answers.add(j+1);
				}
			}
		}
		return answers;
	}
	public static void main(String args[]){
		int bob=6/3;
		System.out.println(bob);
		Scanner kevinisdumb=new Scanner(System.in);
		for(int i=0; i<5; i++){
			String operations=kevinisdumb.next();
			ArrayList<Integer> answers=new ArrayList<Integer>();
			answers=getPositions(operations);
			Collections.sort(answers);
			for(int j=0; j<answers.size(); j++){
				System.out.print(answers.get(j)+" ");
			}
			System.out.println();
		}
		kevinisdumb.close();
	}
}

File: enclosure/src/enclosure.cpp
#include <map>
#include <set>
#include <list>
#include <cmath>
#include <ctime>
#include <deque>
#include <queue>
#include <stack>
#include <string>
#include <bitset>
#include <cstdio>
#include <limits>
#include <vector>
#include <climits>
#include <cstring>
#include <cstdlib>
#include <fstream>
#include <numeric>
#include <sstream>
#include <iostream>
#include <algorithm>
#include <stdlib.h>
#include <map>
using namespace std;

vector<int> getPositions(string operations){\
	vector<int> answers;
	bool open=false;
	int length=operations.length(), position;
	for(int i=0; i<length; i++){
		if(operations.at(i)=='('){
			open=true;
			position=i;
			break;
		}else if(operations.at(i)==')'){
			position=i;
			break;
		}
	}
	if(open){
		for(int j=position+2; j<length; j++){
			if(operations.at(j)!='+'&&operations.at(j)!='-'&&operations.at(j)!='/'&&operations.at(j)!='*'){
				answers.push_back(j+2);
			}
		}
	}else{
		for(int j=position-2; j>=0; j--){
			if(operations.at(j)!='+'&&operations.at(j)!='-'&&operations.at(j)!='/'&&operations.at(j)!='*'){
				answers.push_back(j+1);
			}
		}
	}
	return answers;
}
int main() {
	ifstream kevinisstupid;
	kevinisstupid.open("retard.txt");
	for(int i=0; i<5; i++){
		string operations;
		kevinisstupid>>operations;
		vector<int> answers=getPositions(operations);
		sort(answers.begin(), answers.end());
		int size=answers.size();
		int count=1;
		for(vector<int>::iterator itr=answers.begin(); itr!=answers.end(); itr++){
			cout<<*itr;
			if(count<size){
				cout<<", ";
			}
			count++;
		}
		cout<<endl;
	}

}
